File: CS/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils.Win;
using DevExpress.XtraEditors;

namespace PopupContainerWithOKButton
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }

        private void checkEditShowCustomButton_Properties_EditValueChanged(object sender, EventArgs e)
        {
            popupContainerEditCustomButtonForTest.Properties.ShowCustomButton = (bool)checkEditShowCustomButton.EditValue;
            Control currentPopupWindow = (popupContainerEditCustomButtonForTest as IPopupControl).PopupWindow;
            if (currentPopupWindow != null) currentPopupWindow.Dispose();
        }

        private void MainForm_Load(object sender, EventArgs e)
        {
            checkEditShowCustomButton.EditValue = true;
        }


        private void popupContainerEditCustomButtonForTest_Properties_QueryResultValue(object sender, DevExpress.XtraEditors.Controls.QueryResultValueEventArgs e)
        {
            e.Value = buttonEditSample.EditValue;
        }

        private void popupContainerEditCustomButtonForTest_Properties_CustomButtonClick(object sender, PopupContainerCustomButtonControl.EventsArgsCustomButton e)
        {
            if (e.CustomButton.Text == "Ok")
                label1.Text = "Status: button Ok is pressed";
            else if (e.CustomButton.Text == "Close")
                label1.Text = "Status: button Close is pressed";
            else
                label1.Text = "Status: " + e.CustomButton.Tag.ToString();
        }
    }
}

File: VB/MainForm.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Utils.Win

Namespace PopupContainerWithOKButton

    Public Partial Class MainForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub checkEditShowCustomButton_Properties_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            popupContainerEditCustomButtonForTest.Properties.ShowCustomButton = CBool(checkEditShowCustomButton.EditValue)
            Dim currentPopupWindow As Control = TryCast(popupContainerEditCustomButtonForTest, IPopupControl).PopupWindow
            If currentPopupWindow IsNot Nothing Then currentPopupWindow.Dispose()
        End Sub

        Private Sub MainForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            checkEditShowCustomButton.EditValue = True
        End Sub

        Private Sub popupContainerEditCustomButtonForTest_Properties_QueryResultValue(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.QueryResultValueEventArgs)
            e.Value = buttonEditSample.EditValue
        End Sub

        Private Sub popupContainerEditCustomButtonForTest_Properties_CustomButtonClick(ByVal sender As Object, ByVal e As PopupContainerCustomButtonControl.EventsArgsCustomButton)
            If Equals(e.CustomButton.Text, "Ok") Then
                label1.Text = "Status: button Ok is pressed"
            ElseIf Equals(e.CustomButton.Text, "Close") Then
                label1.Text = "Status: button Close is pressed"
            Else
                label1.Text = "Status: " & e.CustomButton.Tag.ToString()
            End If
        End Sub
    End Class
End Namespace
